roachtest: initialize prometheus scrapers by default for tpc-c

otan opened the pull request:
This commit enables TPC-C prometheus scraping by default. All cockroach
nodes and the workload instance will dump the prometheus snapshot into a
zip at the end of the test run.

This is a "trial by fire" - I expect nothing to go wrong, but if it does
we've caught it right at a branch cut.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/70103)
<!-- Reviewable:end -->

tbg reviewed (approved):
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70103)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @stevendanna)

___
*[pkg/cmd/roachtest/tests/tpcc.go, line 185 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/70103#-MjSw_gx74-4mvA0wDo5:-MjSw_gx74-4mvA0wDo6:b-6oa3re) ([raw file](https://github.com/cockroachdb/cockroach/blob/83ab61a2e2e2d5117830c04940e69f60387f22d7/pkg/cmd/roachtest/tests/tpcc.go#L185)):*
> ```Go
> 	promCfg, cleanupFunc := setupPrometheus(ctx, t, c, opts)
> 	defer cleanupFunc()
> 	if promCfg != nil && opts.ChaosEventsProcessor != nil {
> ```

Should this error out if we have a ChaosEventsProcessor but no promCfg? That would seem like a misconfiguration at this point.


<!-- Sent from Reviewable.io -->

otan committed:
roachtest: initialize prometheus scrapers by default for tpc-c

This commit enables TPC-C prometheus scraping by default. All cockroach
nodes and the workload instance will dump the prometheus snapshot into a
zip at the end of the test run.

This is a "trial by fire" - I expect nothing to go wrong, but if it does
we've caught it right at a branch cut.

Release note: None

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70103)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/cmd/roachtest/tests/tpcc.go, line 185 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/70103#-MjSw_gx74-4mvA0wDo5:-MjVipV13jdDIfZcpFxg:b-7rl0jq) ([raw file](https://github.com/cockroachdb/cockroach/blob/83ab61a2e2e2d5117830c04940e69f60387f22d7/pkg/cmd/roachtest/tests/tpcc.go#L185)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Should this error out if we have a ChaosEventsProcessor but no promCfg? That would seem like a misconfiguration at this point.
</blockquote></details>

done!


<!-- Sent from Reviewable.io -->

otan commented:
bors r=tbg
